R/item_rm_recursive.R

```diff
@@ -1,64 +1,69 @@
-#' Remove an item completely by recursively removing its children
-#'   
-#' @export
+#' @title Remove an item completely by recursively removing its child items
+#' 
+#' @keywords internal
 #' @param id A ScienceBase ID or something that can be coerced to a SB item ID
 #' @param ... Additional parameters are passed on to \code{\link[httr]{GET}} and
 #'   \code{\link[httr]{DELETE}}
 #' @param session Session object from \code{\link{authenticate_sb}}
 #' @return \code{TRUE} to indicate success
-#' @details BEWARE: This removes all folders/files in an item completely.
+#' @details BEWARE: This completely removes ALL CHILD ITEMS AND THEIR CHILDREN
+#'   as well as the item itself.
 #' @examples \dontrun{
 #' # Create an item with nested structure
-#' fname <- "couch"
-#' fold <- folder_create(user_id(), name = fname)
-#' folder_create(fold$id, name = "one")
-#' folder_create(fold$id, name = "two")
-#' df2 <- item_list_children(fold$id)
+#' authenticate_sb()
+#' fname <- "chairs"
+#' folder_create(user_id(), name = fname)
+#' df <- item_list_children(user_id()) # may need to wait a moment to run this
+#' id <- df[ df$title == fname, "id" ]
+#' folder_create(id, name = "one")
+#' folder_create(id, name = "two")
+#' df2 <- item_list_children(id)
```

Comment:
other than that, everything looks good to me on read-through. i don't have time to do real testing tonight but will check out the functionality next time i'm on sbtools. In the meantime, I'm OK with merging if you are. thanks for taking on the merging, @lawinslow